lib/cli/run-helpers.js

```diff
@@ -204,6 +204,7 @@ exports.singleRun = (mocha, {files = [], exit = false} = {}) => {
     // Instead of `process.exit(130)`, set runner.failures to 130 (exit code for SIGINT)
     // The amount of failures will be emitted as error code later
     runner.failures = 130;
+    setImmediate(() => process.kill(process.pid, 'SIGINT'));
```

Comment:
This fires the signal recursively?

It doesn't work for me, FWIW...

Reply:
If you read the GNU signal handling link I posted, reraising the signal is correct.

Reply:
@boneskull 
@plroebuck  -  Updated to reraise the signal once as specified in the comment

Reply:
@plroebuck That's fine, but nowhere does it say you're supposed to trigger your custom *handler* again.  Which is what's happening if you debug this.  ... at least on my machine.

Reply:
@Jayasankar-m has addressed this in 956dc59, but I have [a suggestion](https://github.com/mochajs/mocha/pull/3602#pullrequestreview-184921208)

Reply:
alright, I apologize, this is coming back to me now.

The intent here is to cleanly exit, which it seems we just ditched with the PR.

We call `runner.abort()`, which simply sets a flag in the runner.  *Once the current test completes*, the runner encounters this flag, and exits the process with an exit code reflecting the number of test failures.

This, of course, conflicts with the intent of *this* PR, which was to *not* let the currently-running test complete, and just exit.

I made this change so long ago, I'm not sure about the reasoning behind the "hack"; pretty sure that was me.  Nor am I sure of the reasons behind the intent of a clean exit, but that's the intent nonetheless.

It's likely we've broken the "exit code with # of failures" feature here (which I realize you do not like), but only if we interrupt the process.  

This seems OK to me, not recalling the original reasons for doing this, but *also* means the handler itself is useless and should be removed.

Reply:
OK, it comes from #2438 by @Munter.

We're exiting with `130` to ensure the expected behavior--exiting with code `130`--is retained while the handler disrupts the default behavior.

But because we're now actually invoking the default behavior, we don't need to call `runner.abort()` (because the process will immediately exit), and we *don't* need to set `runner.failures` (because the process will immediately exit) and we *don't* need to set `process.exitCode` if we just remove the damn `SIGINT` handler.

I'll send a PR to this effect.